apps/nerves_hub_web_core/lib/nerves_hub_web_core/firmwares.ex

```diff
@@ -515,32 +525,37 @@ defmodule NervesHubWebCore.Firmwares do
     end
   end
 
+  @typep metadata_string() :: String.t()
+  @typep metadata_key() :: String.t()
+  @typep metadata_value() :: String.t() | nil
+
+  @spec get_fwup_metadata(Path.t()) :: {:ok, metadata_string()} | {:error, String.t()}
   defp get_fwup_metadata(filepath) do
-    case System.cmd("fwup", ["-m", "-i", filepath]) do
+    case System.cmd("fwup", ["-m", "-i", IO.iodata_to_binary(filepath)]) do
```

Comment:
```suggestion
    case System.cmd("fwup", ["-m", "-i", filepath]) do
```

If the spec is correct, then this should be fine.